File: ass-ade-v1.1/tests/fixtures/rebuilt-out/bridges/kotlin/src/main/kotlin/a1_at_functions/ManifestLoader.kt
import java.nio.file.Files
import java.nio.file.Path

private fun extractString(payload: String, prefix: String): String =
    payload.substringAfter(prefix, missingDelimiterValue = "").substringAfter('"', "").substringBefore('"', "")

private fun extractBoolean(payload: String, prefix: String): Boolean =
    payload.substringAfter(prefix, missingDelimiterValue = "").trimStart().startsWith("true")

private fun extractStringArray(payload: String, prefix: String): List<String> {
    val segment = payload.substringAfter(prefix, missingDelimiterValue = "")
        .substringBefore(']', missingDelimiterValue = "")
    if (segment.isEmpty()) {
        return emptyList()
    }
    return segment.split('"').drop(1).filterIndexed { index, _ -> index % 2 == 0 }
}

fun findRepoRoot(start: Path, manifestRel: String = DEFAULT_MANIFEST_REL): Path {
    var current = start
    while (true) {
        if (Files.isRegularFile(current.resolve(manifestRel))) {
            return current
        }
        val parent = current.parent ?: error("Could not find $manifestRel from $start")
        current = parent
    }
}

fun loadManifestPayload(start: Path, manifestRel: String = DEFAULT_MANIFEST_REL): String {
    val repoRoot = findRepoRoot(start, manifestRel)
    return Files.readString(repoRoot.resolve(manifestRel))
}

fun loadBridgeManifest(start: Path): BridgeManifestSummary {
    val payload = loadManifestPayload(start)
    return BridgeManifestSummary(
        schema = extractString(payload, ""schema": "),
        bridgeReady = extractBoolean(payload, ""bridge_ready": "),
        bridgeLanguages = extractStringArray(payload, ""bridge_languages": ["),
        pythonBridgeCommand = extractStringArray(payload, ""python_bridge_command": ["),
    )
}

File: ass-ade-v1.1/tests/fixtures/rebuilt-out/bridges/typescript/src/a1_at_functions/manifest_loader.ts
import { readFileSync } from "node:fs";
import { dirname, join } from "node:path";

import { DEFAULT_MANIFEST_REL, type BridgeManifest } from "../a0_qk_constants/bridge_contract.js";

function asStringArray(value: unknown): string[] {
  return Array.isArray(value) ? value.map((item) => String(item)) : [];
}

export function findRepoRoot(start: string = process.cwd(), manifestRel: string = DEFAULT_MANIFEST_REL): string {
  let current = start;
  for (;;) {
    const candidate = join(current, manifestRel);
    try {
      readFileSync(candidate, "utf-8");
      return current;
    } catch (error) {
      void error;
    }
    const parent = dirname(current);
    if (parent === current) {
      throw new Error(`Could not find ${manifestRel} from ${start}`);
    }
    current = parent;
  }
}

export function loadBridgeManifest(root: string = process.cwd(), manifestRel: string = DEFAULT_MANIFEST_REL): BridgeManifest {
  const repoRoot = findRepoRoot(root, manifestRel);
  const payload = JSON.parse(readFileSync(join(repoRoot, manifestRel), "utf-8")) as Record<string, unknown>;
  return {
    schema: String(payload.schema ?? ""),
    bridge_ready: Boolean(payload.bridge_ready),
    bridge_languages: asStringArray(payload.bridge_languages),
    python_bridge_command: asStringArray(payload.python_bridge_command),
    supported_languages: asStringArray(payload.supported_languages),
  };
}
